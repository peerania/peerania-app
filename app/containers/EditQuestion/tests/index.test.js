import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import configureStore from 'configureStore';
import createdHistory from 'createdHistory';

import EosioProvider from 'containers/EosioProvider';
import { EditQuestion } from '../index';

const child = <div>children</div>;
React.Children.only = jest.fn().mockImplementation(() => child);

// const cmp = new EditQuestion();
const props = {
  match: {
    params: {
      questionid: 1,
      answerid: 1,
    },
  },
  locale: 'en',
  questionLoading: false,
  question: {
    title: 'title',
    content: 'content',
  },
  getAskedQuestionDispatch: jest.fn(),
  editQuestionDispatch: jest.fn(),
};

describe('EditQuestion', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, createdHistory);
  });

  /*describe('componentDidMount', () => {
    cmp.componentDidMount();

    expect(cmp.props.getAskedQuestionDispatch).toHaveBeenCalledWith(
      cmp.props.match.params.questionid,
    );
  });

  describe('editQuestion', () => {
    const values = new Map();
    cmp.editQuestion(values);

    expect(cmp.props.editQuestionDispatch).toHaveBeenCalledWith(
      {
        title: values.get('FORM_TITLE'),
        content: values.get('FORM_CONTENT'),
      },
      cmp.props.match.params.questionid,
    );
  });*/

  describe('should render and match the snapshot', () => {
    it('questionLoading is falsy', () => {
      props.questionLoading = false;
      const {
        container: { firstChild },
      } = render(
        <Provider store={store}>
          <EosioProvider>
            <EditQuestion {...props} />
          </EosioProvider>
        </Provider>,
      );
      expect(firstChild).toMatchSnapshot();
    });

    it('questionLoading is true', () => {
      props.questionLoading = true;
      const {
        container: { firstChild },
      } = render(
        <Provider store={store}>
          <EosioProvider>
            <EditQuestion {...props} />
          </EosioProvider>
        </Provider>,
      );
      expect(firstChild).toMatchSnapshot();
    });

    it('question is null', () => {
      props.question = null;
      const {
        container: { firstChild },
      } = render(
        <Provider store={store}>
          <EosioProvider>
            <EditQuestion {...props} />
          </EosioProvider>
        </Provider>,
      );
      expect(firstChild).toMatchSnapshot();
    });

    it('question is NOT null', () => {
      props.question = {
        title: 'title',
        content: 'content',
      };
      const {
        container: { firstChild },
      } = render(
        <Provider store={store}>
          <EosioProvider>
            <EditQuestion {...props} />
          </EosioProvider>
        </Provider>,
      );
      expect(firstChild).toMatchSnapshot();
    });
  });
});
