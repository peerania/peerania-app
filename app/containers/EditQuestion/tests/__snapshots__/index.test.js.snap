// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditQuestion should render and match the snapshot question is NOT null 1`] = `
<div>
  children
</div>
`;

exports[`EditQuestion should render and match the snapshot question is null 1`] = `
<div>
  children
</div>
`;

exports[`EditQuestion should render and match the snapshot questionLoading is falsy 1`] = `
<div>
  children
</div>
`;

exports[`EditQuestion should render and match the snapshot questionLoading is true 1`] = `
<div>
  children
</div>
`;
